allow suffix after a mail address for RFC 3461

my-name-is-already-taken opened the pull request:
This change makes it possible to Append a DSN specification behind the MAIL and RCPT mail addresses in the smtp protocol.

Currently, curl checks if these addresses start with a '<' or end with a '>' and remove these characters, before adding them when issuing the actual command. So when setting `CURLOPT_MAIL_RCPT` to `<foo@bar.com> NOTIFY=FAILURE`, the actual address sent to the server becomes `<foo@bar.com> NOTIFY=FAILURE>`.

This proposal changes the behavior in the following way: If the address starts with a '<' character, we search the address for the last '>' instead of only checking the last character. Everything that comes behind this bracket is then kept as a suffix in the address.

See also https://github.com/curl/curl/issues/8232.

testclutch commented:
Analysis of PR #16643 at [fafb6ae0](https://github.com/curl/curl/commit/fafb6ae0ae9e013f0a8966cbc4efe1995b47e3e9):

[Test 1515 failed](https://github.com/curl/curl/actions/runs/13760369169/job/38475029105#check-step-15), which has NOT been flaky recently, so **there could be a real issue in this PR**.

###### Generated by [Testclutch](https://testclutch.curl.se/)

bagder commented:
This looks like a good idea to support. I would just like to add for a little more here:

- documentation - this needs to be described
- a test case or two that verify that this behavior works as intended and documented

my-name-is-already-taken commented on lib/smtp.c:
```diff
@@ -654,11 +657,12 @@ static CURLcode smtp_perform_mail(struct Curl_easy *data)
     if(data->set.str[STRING_MAIL_AUTH][0] != '\0') {
       char *address = NULL;
       struct hostname host = { NULL, NULL, NULL, NULL };
+      char const *suffix = "";
```
Fixed in e06201b

my-name-is-already-taken commented:
I have fixed an error and added a testcase as well as some documentation.

I had difficulties writing the test, so I would appreciate it if someone checked whether I've done it correcty.

Please let me know if something else is missing.

vszakats commented:
Spellcheck says:
```
Misspelled words:
<htmlcontent> docs/libcurl/opts/CURLOPT_MAIL_RCPT.md: html>body>p
--------------------------------------------------------------------------------
DSN
--------------------------------------------------------------------------------
```diff
This should silence it:
--- i/.github/scripts/spellcheck.words
+++ w/.github/scripts/spellcheck.words
@@ -203,6 +203,7 @@ DoT
 doxygen
 drftpd
 dsa
+DSN
 Dudka
 Dymond
 dynbuf
```

vszakats reviewed (approved):
LGTM
